apps/internal/base/base.go

```diff
@@ -345,6 +350,26 @@ func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilen
 	if silent.Claims == "" {
 		ar, err = AuthResultFromStorage(storageTokenResponse)
 		if err == nil {
+			if shouldRefresh(storageTokenResponse.AccessToken.RefreshOn.T) {
+				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Why 10 seconds instead of obeying the general context cancellation? Since refresh happens on the same thread as the request, it makes sense to follow normal rules there.